### PR TITLE
add command to (slowly) prune lotus chain datastore

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
Decent example usage:
```
./lotus-shed state-prune --delete-up-to=5000000 --use-bloom-set
```

Note: the gc part at the end takes a loooong time and doesnt really seem to clear everything up. You can run:
```
./lotus-shed state-prune --only-ds-gc
```
to just run badger gc in a loop for 20 times.

File: cmd/lotus-shed/pruning.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+var stateTreePruneCmd = &cli.Command{
+	Name:        "state-prune",
+	Description: "Deletes old state root data from local chainstore",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "repo",
+			Value: "~/.lotus",
+		},
```
Comment (travisperson) [inline cmd/lotus-shed/pruning.go:92]: This is a global flag: https://github.com/filecoin-project/lotus/blob/master/cmd/lotus-shed/main.go#L46-L51

**Comment by magik6k** [comment]
lint sad

**Review by magik6k — APPROVED** [review]

**Comment by s0nik42** [comment]
Hello ! Is this safe to use compared to the previous proposition of @ribasushi  to delete the chain content folder and import the pruned version of the chain ?

Thank you ! 

Julien

**Pull request merged**